R/qMeth.R

```diff
@@ -24,337 +24,554 @@
+#'   Valid values are between 0 and 255. The default (255) will include
+#'   all alignments.
+#' @param clObj A cluster object to be used for parallel processing of
+#'   multiple files (see \sQuote{Details}).
+#' 
+#' @export
+#' 
+#' @return 
+#' For \code{reportLevel}=\dQuote{C}, a \code{GRanges} object if
+#' \code{asGRanges}=\code{TRUE}, otherwise a \code{data.frame}.
+#' 
+#' Each row contains the coordinates of individual cytosines for
+#' \code{collapseByQueryRegion}=\code{FALSE} or query regions
+#' for \code{collapseByQueryRegion}=\code{TRUE}.
+#' 
+#' In addition to the coordinates columns (or \code{seqnames},
+#' \code{ranges} and \code{strand} slots for \code{GRanges} objects),
+#' each row contains per bam file:
+#' 
+#' Two values (total and methylated events, with suffixes _T and _M), or
+#' if the \code{qProject} object was created including a SNP table,
+#' six values (total and methylated events for Reference, Unknown and
+#' Alternative genotypes, with suffixed _TR, _TU, _TA, _MR, _MU and _MA).
+#' In the latter case, C's or CpG's that overlap with SNPs in the table
+#' are removed.
+#' 
+#' If \code{collapseBySample}=\code{TRUE}, groups of bam files with
+#' identical sample name are combined (summed) and will be represented by
+#' a single set of total and methylated count columns.
+#' 
+#' If \code{mode}=\dQuote{var}, the _T and _M columns correspond to total
+#' and matching alignments overlapping the guanine paired to the cytosine.
+#' 
+#' For \code{reportLevel}=\dQuote{alignment}, a \code{list} with one
+#' element per bam file or sample (depending on \code{collapseBySample}). 
+#' Each list element is another list with the elements:
+#' \itemize{
+#'   \item \code{aid}: character vector with unique alignment identifiers
+#'   \item \code{Cid}: integer vector with genomic coordinate of C base
+#'   \item \code{strand}: character vector with the strand of the C base
+#'   \item \code{meth}: integer vector with methylation state for
+#'   alignment and C defined by \code{aid} and \code{Cid}. The values are
+#'   1 for methylated or 0 for unmethylated states.
+#' }
+#' 
+#' @author Anita Lerch, Dimos Gaidatzis and Michael Stadler
+#' @keywords utilities misc
+#' 
+#' @seealso 
+#' \code{\link[QuasR]{qAlign}},
+#' \code{\link[parallel]{makeCluster}} from package \pkg{parallel}
+#' 
+#' @name qMeth
+#' @aliases qMeth
+#' 
+#' @examples 
+#' # copy example data to current working directory
+#' file.copy(system.file(package="QuasR", "extdata"), ".", recursive=TRUE)
+#' 
+#' # create alignments
+#' sampleFile <- "extdata/samples_bis_single.txt"
+#' genomeFile <- "extdata/hg19sub.fa"
+#' proj <- qAlign(sampleFile, genomeFile, bisulfite="dir")
+#' proj
+#' 
+#' # calculate methylation states
+#' meth <- qMeth(proj, mode="CpGcomb")
+#' meth
+#' 
+#' @importFrom Rsamtools scanBamHeader scanFaIndex
+#' @importFrom parallel clusterEvalQ clusterApplyLB
+#' @importFrom GenomeInfoDb seqlevels seqinfo
```

Comment:
should also import `seqnames` from `GenomeInfoDb`

Reply:
Fixed